File: local-client/maestro_local/i18n.py
"""Internacionalização simples baseada em catálogo PT->EN.

O idioma-fonte do código é o português. `t("texto em pt")` retorna a
tradução em inglês quando o idioma ativo é "en"; caso contrário devolve
o próprio texto. A troca de idioma exige reiniciar o app (as telas são
construídas com o idioma vigente no startup).
"""
from __future__ import annotations

SUPPORTED = {"pt": "Português", "en": "English"}
DEFAULT_LANG = "pt"

_lang = DEFAULT_LANG


def init_language() -> None:
    """Carrega o idioma salvo na config (chamar no startup)."""
    global _lang
    try:
        from maestro_local.config import get_language
        code = get_language()
        if code in SUPPORTED:
            _lang = code
    except Exception:  # noqa: BLE001
        _lang = DEFAULT_LANG


def current_language() -> str:
    return _lang


def set_runtime_language(code: str) -> None:
    global _lang
    if code in SUPPORTED:
        _lang = code


def t(text: str) -> str:
    """Traduz `text` (em pt) para o idioma ativo."""
    if _lang == "pt":
        return text
    from maestro_local.i18n_catalog import CATALOG
    return CATALOG.get(_lang, {}).get(text, text)


def tf(text: str, **kwargs) -> str:
    """Traduz e formata: t(text).format(**kwargs)."""
    return t(text).format(**kwargs)

File: local-client/webui/src/i18n.js
import { CATALOG } from './i18n_catalog'

// Idioma-fonte é o português; t(pt) devolve a tradução quando o idioma é "en".
let _lang = 'pt'

export function setLang(l) {
  _lang = l === 'en' ? 'en' : 'pt'
}
export function getLang() {
  return _lang
}

export function t(s) {
  if (_lang === 'pt') return s
  return (CATALOG[_lang] && CATALOG[_lang][s]) || s
}

// Traduz e formata com {placeholders}: tf("Olá {name}", {name:"x"})
export function tf(s, vars = {}) {
  let out = t(s)
  for (const [k, v] of Object.entries(vars)) out = out.replaceAll(`{${k}}`, v)
  return out
}
